src/partitioning/parmetis_partitioner.C

```diff
@@ -410,6 +416,52 @@ void ParmetisPartitioner::build_graph (const MeshBase& mesh)
   // face neighbors
   const dof_id_type n_active_local_elem  = mesh.n_active_local_elem();
 
+  // If we have boundary elements in this mesh, we want to account for
+  // the connectivity between them and interior elements.  We can find
+  // interior elements from boundary elements, but we need to build up
+  // a lookup map to do the reverse.
+
+  typedef LIBMESH_BEST_UNORDERED_MAP<const Elem *, const Elem *>
+    map_type;
+  map_type interior_to_boundary_map;
+
+  {
+    MeshBase::const_element_iterator       elem_it  = mesh.active_elements_begin();
```

Comment:
I don't think so.  We might have non-local boundary elements that link to local interior elements.